src/Arcomage/Arcomage.cpp

```diff
@@ -240,8 +240,7 @@ void ArcomageGame::onKeyPress(PlatformKey key) {
 explosion_effect_struct *explosion_effect_struct::New() {
     explosion_effect_struct *v2 = (explosion_effect_struct *)malloc(sizeof(explosion_effect_struct));
     if (v2 == nullptr) {
-        logger->Warning("Malloc error");
-        Error("Malloc");  // is this recoverable
+        Error("Malloc - explosion_effect_struct::New()");  // is this recoverable
```

Comment:
Btw I think we shouldn't handle malloc / new failures. If you're out of memory then there is very little you can do, so just crashing shall be OK.